extension/content.js

```diff
@@ -168,13 +168,22 @@
     const turndownService = configureTurndownService(settings);
     
     try {
-      const markdown = turndownService.turndown(content);
+      let markdown = turndownService.turndown(content);
       
       // Validate markdown output
       if (!markdown || markdown.trim() === '') {
         throw new Error('Conversion resulted in empty markdown');
       }
       
+      // Add page title as H1 if enabled and title is non-empty
+      if (settings.includeTitle) {
+        debugger;
+        const pageTitle = document.title.trim();
+        if (pageTitle.length > 0) {
```

Comment:
Good consolidation of title handling logic! I like that you've moved the title addition to a centralized location rather than having it scattered across multiple extraction functions